source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -267,14 +267,17 @@ const std::string HystrixSink::printRollingWindows() {
 
 HystrixSink::HystrixSink(Server::Instance& server, const uint64_t num_buckets)
     : server_(server), current_index_(num_buckets > 0 ? num_buckets : DEFAULT_NUM_BUCKETS),
-      window_size_(current_index_ + 1) {
+      window_size_(current_index_ + 1),
+      cluster_upstream_rq_time_("cluster.upstream_rq_time", server.stats().symbolTable()) {
   Server::Admin& admin = server_.admin();
   ENVOY_LOG(debug,
             "adding hystrix_event_stream endpoint to enable connection to hystrix dashboard");
   admin.addHandler("/hystrix_event_stream", "send hystrix event stream",
                    MAKE_ADMIN_HANDLER(handlerHystrixEventStream), false, false);
 }
 
+HystrixSink::~HystrixSink() { cluster_upstream_rq_time_.free(server_.stats().symbolTable()); }
```

Comment:
I can accept that we want to do non-RAII stuff deep in the stats code, but it seems unnecessary in code like this. Can we have an RAII wrapper here and similar cases? Presumably there are going to be a lot of cases like this as you use this API elsewhere, right?

Reply:
Great point. I renamed StatNameTempStorage to StatNameManagedStorage and used it here, eliminating the need to have a destructor. Also added more commenting for StatNameManagedStorage to suggest using it in these scenarios.

That expanded this PR slightly to include a boring rename.